crates/orderbook/src/database/fee_policies.rs

```diff
@@ -51,16 +59,24 @@ impl super::Postgres {
             .collect::<HashMap<_, _>>();
         timer.stop_and_record();
 
-        fee_policies
-            .into_iter()
-            .map(|((auction_id, order_uid), policies)| {
-                policies
-                    .into_iter()
-                    .map(|policy| fee_policy_from(policy, quotes.get(&order_uid), order_uid))
-                    .collect::<anyhow::Result<Vec<_>>>()
-                    .map(|policies| ((auction_id, order_uid), policies))
-            })
-            .collect::<anyhow::Result<HashMap<_, _>>>()
+        let mut result = HashMap::new();
+        for (key, executed_fees) in executed_protocol_fees {
+            if let Some(policies) = fee_policies.get(&key) {
```

Comment:
I was hesitant about this. So, in practice, this would never happen as we use fee policies to calculate protocol fees in the first place. Opposite could happen, for fee policy to miss executed protocol fee.
Truthfully, ☝️ statement assumes how the rest of the system works right now.
So 🤷‍♂️ , I am fine with both having or not having a log here (or even return err).

Reply:
In that case let's at least throw a log in there to notice when something is weird.
I wouldn't go as far as returning an error, though.

Reply:
added